app/models/account.rb

```diff
@@ -80,12 +99,13 @@ def self.using_journal?
 
   def self.for_facility(facility)
     if facility.single_facility?
-      where(
-        "accounts.type in (:allow_all) or (accounts.type in (:limit_one) and accounts.facility_id = :facility)",
-        allow_all: config.global_account_types,
-        limit_one: config.facility_account_types,
-        facility: facility,
-      )
+      left_outer_joins(:facilities)
+        .where(
+          "accounts.type in (:global_account_types) or (accounts.type in (:facility_account_types) and account_facility_joins.facility_id = :facility)",
+          global_account_types: config.global_account_types,
+          facility_account_types: config.facility_account_types,
+          facility: facility,
+        )
     else
```

Comment:
I think this `where` can be simplified by extracting a scope for global accounts and using Rails’s `or` support. How about something like this?

```
scope :global, -> { where(type: config.global_account_types) }
scope :per_facility -> { where(type: config.facility_account_types) }

def self.for_facility(facility)
  if facility.single_facility?
    global.
    or(per_facility.left_ourter_joins(:facilities).where(account_facility_joins: { facility_id: facility.id }))
  else
```

Reply:
I totally dig that if it works. I'll give it a shot. 

Also, I'm probably going to rename `single_facility?` and `cross_facility?` to something like `per_facility?` (if we think that's a good name) and `global?` since the old names won't really be accurate anymore.

Reply:
I like those names! My code may not work exactly as-is (the `.where` call may need a little more help but I think it’s worth it to use the hash over a raw string), but I think you get the idea.